From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hayanesuru <hayanesuru@outlook.jp>
Date: Wed, 6 Aug 2025 17:12:21 +0900
Subject: [PATCH] Fix elytra kinetic velocity desync


diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 974bfe44d802abb5d80f3bebce487f9225dfbc6a..a7672543689f180100a2422c7f59875ec517c649 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -3265,7 +3265,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
             float f = (float)(d * 10.0 - 3.0);
             if (f > 0.0F) {
                 this.playSound(this.getFallDamageSound((int)f), 1.0F, 1.0F);
-                if (level().purpurConfig.elytraKineticDamage) // Purpur - Toggle for kinetic damage
+                if (level().purpurConfig.elytraKineticDamage) f = 0.0F; // Purpur - Toggle for kinetic damage // Leaf - Fix elytra kinetic velocity desync
                 this.hurt(this.damageSources().flyIntoWall(), f);
             }
         }
